libraries/botbuilder-lg/src/expander.ts

```diff
@@ -264,14 +264,21 @@ export class Expander extends AbstractParseTreeVisitor<unknown[]> implements LGT
             const tempRes: Record<string, unknown>[] = [];
             for (const res of finalResult) {
                 for (const refValue of templateRefValues.get(templateRefValueKey)) {
-                    tempRes.push(
-                        JSON.parse(
-                            JSON.stringify(res).replace(templateRefValueKey, refValue.toString().replace(/"/g, '\\"'))
-                        )
-                    );
+                    const myReg1 = new RegExp(/"/g);
+                    const myReg2 = new RegExp(/\\/g);
+                    let hasBeenReplaced = false;
+                    let currentRefValue = refValue.toString();
+                    if (myReg1.test(currentRefValue)) {
+                        hasBeenReplaced = true;
+                        currentRefValue = currentRefValue.replace(/"/g, '\\"');
+                    }
+                    if (!hasBeenReplaced && myReg2.test(currentRefValue)) {
+                        hasBeenReplaced = true;
+                        currentRefValue = currentRefValue.replace(/\\/g, '\\\\');
+                    }
+                    tempRes.push(JSON.parse(JSON.stringify(res).replace(templateRefValueKey, currentRefValue)));
```

Comment:
This functionality i'll change it a bit, since im looking it will only replace the `\\` if the `"` wasnt replaced, shouldnt be replaced anyway?, even if the `"` was repalced or not?

First
the naming of the two regex can be improved instead of just 1 and 2.

Second
The code can be reduced, first it replaces the `\\` and then the `"`
```suggestion
                    const myReg1 = new RegExp(/"/g);
                    const myReg2 = new RegExp(/\\/g);
                    const currentRefValue = refValue.toString()
                      .replace(myReg2, '\\\\')
                      .replace(myReg1, '\\"');
                    tempRes.push(JSON.parse(JSON.stringify(res).replace(templateRefValueKey, currentRefValue)));
```

or you can use the regex in the replace
```suggestion
                    const currentRefValue = refValue.toString()
                      .replace(/\\/g, '\\\\')
                      .replace(/"/g, '\\"');
                    tempRes.push(JSON.parse(JSON.stringify(res).replace(templateRefValueKey, currentRefValue)));
```

Reply:
yes, it looks better